### PR TITLE
fix: Crash with read receipts system message WPB-4246

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4246" title="WPB-4246" target="_blank">BugWPB-4246</a>  [iOS] Crash when creating read receipt system message
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We have many crash logs in DataDog that identify the code that caused the crash:
```
0x00000001043470a4 Swift runtime failure: Unexpectedly found nil while implicitly unwrapping an Optional value + 0
1   Wire                               
 0x00000001043470a4 specialized ReadReceiptViewModel.createSystemMessage(template: String) + 1236
```

### Causes (Optional)

We force unwrap the system message for read receipts. It seems that for some reason we sometimes don't know the sender's name and don't create the text for the system message. As a result, the system message is nill and we are trying to to force unwrap it.

### Solutions

Do not force unwrap the system message and use `Someone` when we don't have the sender name.

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationReadReceiptSettingChangedCellDescription.swift [hunk 1]
```diff
@@ -30,26 +30,25 @@ struct ReadReceiptViewModel {
         return iconColor.map { icon.makeImage(size: .tiny, color: $0) }
     }
 
-    func createSystemMessage(template: String) -> NSAttributedString {
-        var updateText: NSAttributedString! = .none
+    func createSystemMessage(template: String) -> NSAttributedString? {
+        var updateText: NSAttributedString?
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationReadReceiptSettingChangedCellDescription.swift]: Does this need to be optional if we always give it a value?

Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationReadReceiptSettingChangedCellDescription.swift]: Yes, it doesn't need to be optional in this implementation. 
Fixed

**Review by johnxnguyen — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Review by netbe — APPROVED** [review]
Any idea in which case this happens @KaterinaWire ? the message does not come from no one, someone did send the receipt

**Comment by KaterinaWire** [comment]
> Any idea in which case this happens @KaterinaWire ? the message does not come from no one, someone did send the receipt

@netbe  I don't know and cannot reproduce it. But we have many logs about the same crash in DataDog every day.

**Pull request merged**